### PR TITLE
add support for using polyglot on language detection

### PR DESCRIPTION
**Pull request by mugurrus**

### CODE REVIEW

File: snoop/defaultsettings.py [hunk 1]
```diff
@@ -88,3 +89,5 @@
 celery.app.conf.task_routes = {
     'snoop.data.tasks.check_if_idle': {'queue': 'watchdog'}
 }
+
+LANGUAGE_DETECTOR = language_detection.detectors[LANGUAGE_DETECTOR_NAME]
```
Comment (mgax) [inline snoop/defaultsettings.py]: I think the django settings should only contain simple values, not references to python objects. I'd move this code to `snoop/data/indexing.py`.

File: requirements.in [hunk 2]
```diff
@@ -14,3 +14,6 @@ langdetect
 django-silk
 honcho
 flower
+polyglot
+pyicu
+pycld2
```
Comment (mgax) [inline requirements.in]: Not for this PR, but we should consider [pipenv](https://pipenv.readthedocs.io/en/latest/), it's much nicer to use than virtualenv+pip-tools.

File: snoop/data/language_detection/polyglot.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
+from polyglot.detect import Detector
+
+
+NAME='polyglot'
+
+def detect(txt):
+    detector = Detector(txt)
+    if detector.language.code and detector.language.code != 'un' and \
```
Comment (mgax) [inline snoop/data/language_detection/polyglot.py]: Please avoid having to split long lines with `\`. You can extract `detector.language.code` into a variable and make the whole function more readable.

File: snoop/defaultsettings.py (hunk 1 above)
Comment (mugurrus) [reply snoop/defaultsettings.py]: fixed

**Pull request merged**